### ISSUE TITLE
There is a RuntimeError "You must compile your model before using it" when I train with resume which has trained 19000 steps.

### ISSUE THREAD

**Issue by maliyong** [fizyr/keras-retinanet#1005]
There is a RuntimeError "You must compile your model before using it" when I train with resume which has trained 19000 steps.
Is anyone meet the problem?  Look for your help.

**Comment by martinzlocha** [comment]
There is a lot of information missing, what parameters are you setting when training, have you modified the code in any way?

**Comment by hgaiser** [comment]
@maliyong can you provide more information?

**Comment by xliu79** [comment]
I also met the same problem.... Does anyone already solved???

**Comment by hgaiser** [comment]
@xliu79 can you provide more information?

**Comment by xliu79** [comment]
I tried to resume the training. But when I run python3 keras_retinanet/bin/train.py --snapshot previous_model.h5 csv my_annotation.csv my_class.csv, I got that error.Here are the screenshots below.
![problem1](https://user-images.githubusercontent.com/31554245/60571765-c498f000-9da6-11e9-90a2-a1cd6c7bfeed.png)
![problem2](https://user-images.githubusercontent.com/31554245/60571939-117cc680-9da7-11e9-93f1-a46f5d4f8c23.png)
@hgaiser

**Comment by hgaiser** [comment]
What was the command you used for training? You can't resume a multi-gpu trained model.

**Comment by xliu79** [comment]
ops. I trained the previous model by multi-gpu. So I can't resume the training anymore?

**Comment by hgaiser** [comment]
You can, but not with `--snapshot`. You can "resume" training with `--weights` instead. The difference being that `--weights` doesn't restore the state of the optimizer (so learning rate, momentum, etc.).

I'm closing this issue as this was most likely the case of @maliyong as well and there has been no further response from him.

**Comment by xliu79** [comment]
Thanks @hgaiser

**Comment by ArthurVil** [comment]
Same problem on single GPU. Resuming the training from the weights works fine, but I get the same error if using snapshots. 
Also I tried to (re)compile the model after loading the snapshot, then the training restart from the epoch of the snapshot, but it seems that it actually restart from scratch es the loss is very high. (Note the lr have been constant all over initial training)
